org.lflang.tests/src/org/lflang/tests/compiler/MixedRadixIntTest.java

```diff
@@ -0,0 +1,113 @@
+        try {
+            new MixedRadixInt(digits, radixes, null);
+        } catch (IllegalArgumentException ex) {
+            Assertions.assertTrue(ex.getMessage().startsWith("Invalid"));
+            return;
+        }
+        Assertions.assertTrue(false, "Expected exception did not occur.");
+    }
+    
+    @Test
+    public void createWithNullAndSet() throws Exception {
+        MixedRadixInt num = new MixedRadixInt(null, radixes, null);
+        Assertions.assertEquals(0, num.get());
+        Assertions.assertEquals("0%2", num.toString());
+        num.set(1 + 2*2 + 3*6 + 4*24);
+        Assertions.assertEquals(1 + 2*2 + 3*6 + 4*24, num.get());
+        int mag = num.magnitude();
+        Assertions.assertEquals(1 + 2*2 + 3*6 + 4*24, mag);
+        num.increment(); // wrap to zero.
+        Assertions.assertEquals(0, num.get());
+        Assertions.assertEquals(0, num.magnitude());
+        
+        num = new MixedRadixInt(null, radixes, null);
+        num.setMagnitude(mag);
+        Assertions.assertEquals(mag, num.magnitude());
+    }
+    
+    @Test
+    public void testPermutation() throws Exception {
+        List<Integer> radixes = new ArrayList<Integer>(List.of(2, 5));
+        List<Integer> digits = new ArrayList<Integer>(List.of(1, 2));
+        List<Integer> permutation = new ArrayList<Integer>(List.of(1, 0));
+        MixedRadixInt num = new MixedRadixInt(digits, radixes, permutation);
+        Assertions.assertEquals(5, num.get());
+        Assertions.assertEquals(2, num.get(1));
+        Assertions.assertEquals(7, num.magnitude());
+        num.increment();
+        Assertions.assertEquals(7, num.get());
+        Assertions.assertEquals(8, num.magnitude());
+        
+        num = new MixedRadixInt(null, radixes, permutation);
+        num.setMagnitude(8);
+        Assertions.assertEquals(8, num.magnitude());
+        Assertions.assertEquals(7, num.get());
+
+        // Test radix infinity digit (effectively).
+        digits = new ArrayList<Integer>(List.of(1, 2, 1));
+        radixes = new ArrayList<Integer>(List.of(2, 5, 1000000));
+        num = new MixedRadixInt(digits, radixes, null);
+        Assertions.assertEquals(15, num.get());
+        Assertions.assertEquals(7, num.get(1));
+        Assertions.assertEquals(15, num.magnitude());
+        
+        permutation = new ArrayList<Integer>(List.of(1, 0, 2));
+        num = new MixedRadixInt(digits, radixes, permutation);
+        num.increment();
+        Assertions.assertEquals(17, num.get());
+        Assertions.assertEquals(18, num.magnitude());
+
+        num = new MixedRadixInt(null, radixes, permutation);
+        num.setMagnitude(18);
+        Assertions.assertEquals(18, num.magnitude());
+        Assertions.assertEquals(17, num.get());
+    }
+    
+    @Test
+    public void testIncrement() throws Exception {
+        List<Integer> radixes = new ArrayList<Integer>(List.of(2, 3));
+        List<Integer> digits = new ArrayList<Integer>(List.of(0, 2));
```

Comment:
I get java.lang.UnsupportedOperationException with this change, so I'm not committing this.

Reply:
That's probably because you're mutating the data structure elsewhere. The better fix would be to eliminate those mutations rather than construct a immutable list and then making a mutable copy of it. See: [UnsupportedOperationException ](https://docs.oracle.com/javase/9/docs/api/java/util/List.html). If you really need a mutable list, then use `Arrays.asList(0, 2);`